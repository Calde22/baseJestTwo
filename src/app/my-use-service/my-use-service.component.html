<h1>Usado el servicio</h1>

<div *ngIf="!wcm_data; else elseBlock">
    <h4>Que servicio tan lento Ome!!!x</h4>
</div>
<ng-template #elseBlock>
    {{wcm_data | json}}
</ng-template>