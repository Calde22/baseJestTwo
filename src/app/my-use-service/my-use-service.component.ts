import { Component, OnInit } from '@angular/core';
import { WcmTdc } from '../interfaces';
import { WcmService } from '../services/wcm.service';

@Component({
  selector: 'app-my-use-service',
  templateUrl: './my-use-service.component.html',
  styleUrls: ['./my-use-service.component.scss']
})
export class MyUseServiceComponent implements OnInit {

  public wcm_data?: WcmTdc

  constructor(private wcm_service: WcmService) { }

  ngOnInit(): void {
    this.wcm_service.getWcm().subscribe( get_wcm_data =>{
      this.wcm_data = get_wcm_data
      console.log(get_wcm_data)
    })
  }

}
