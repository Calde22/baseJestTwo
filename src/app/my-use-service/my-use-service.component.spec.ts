import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MyUseServiceComponent } from './my-use-service.component';
import { HttpClientTestingModule,HttpTestingController } from '@angular/common/http/testing'


import { WcmService } from '../services/wcm.service';

describe('MyUseServiceComponent', () => {
  let component: MyUseServiceComponent;
  let fixture: ComponentFixture<MyUseServiceComponent>;
  //Las variables
  let compiled: HTMLElement;
  let service_wcm: WcmService;
  let httpMock : HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [MyUseServiceComponent],
      imports: [HttpClientTestingModule],
      providers: [WcmService]
    })
      .compileComponents();

    fixture = TestBed.createComponent(MyUseServiceComponent);
    component = fixture.componentInstance;
    service_wcm = TestBed.inject(WcmService)
    httpMock = TestBed.inject(HttpTestingController)
    
    fixture.detectChanges();
    compiled = fixture.nativeElement
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  })

  it('Esto es una foto como la ves 🖼️ snapshot', ()=>{
    //compiled.innerHTML
    expect(compiled).toMatchSnapshot()
  })

  it('Verificando el Loading.... ⏲️', ()=>{
    const h4 = compiled.querySelector('h4')
    expect(h4?.textContent).toContain('Que servicio tan lento Ome!!!')
  })

  it('Inyectar Data al servicio', () =>{
    const mockdata = {}
    const my_urService = "https://www.bancolombia.com/caass?current=true&urile=wcm:path:bancolombiapersonas/asTarjetadeCredito/contServFilterTdc&mime-type=application/json"
    const request = httpMock.expectOne(my_urService)

    expect(request.request.method).toBe('GET')
    request.flush(mockdata)

    fixture.detectChanges()

  })

});
