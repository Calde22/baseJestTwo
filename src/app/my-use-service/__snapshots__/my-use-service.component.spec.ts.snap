// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyUseServiceComponent Esto es una foto como la ves üñºÔ∏è snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Usado el servicio
  </h1>
  <div>
    <h4>
      Que servicio tan lento Ome!!!x
    </h4>
  </div>
  
  
</div>
`;
